builder/bundles.go

```diff
@@ -363,14 +371,32 @@ func resolvePackages(numWorkers int, set bundleSet, packagerCmd []string, emptyD
 	return &bundleRepoPkgs
 }
 
-func installFilesystem(packagerCmd []string, chrootDir string) error {
-	installArgs := merge(packagerCmd,
-		"--installroot="+chrootDir,
-		"install",
-		"filesystem",
-	)
-	_, err := helpers.RunCommandOutputEnv(installArgs[0], installArgs[1:], []string{"LC_ALL=en_US.UTF-8"})
-	return err
+func installFilesystem(chrootDir string, localPath string, bundle *bundle, packagerCmd []string, downloadRetries int) error {
+	var rpmFull string
+	var rpm string
+	for rpm = range bundle.AllRpmPackages {
+		if strings.Contains(rpm, "filesystem") {
+			rpmFull = filepath.Join(localPath, rpm)
+			rpmMap[rpm] = true
+			break
+		}
+	}
+
+	if _, err := os.Stat(rpmFull); os.IsNotExist(err) {
+		if !Offline {
+			rpmPath, err := downloadRpm(packagerCmd, []string{"filesystem"}, chrootDir, downloadRetries)
+			if err != nil {
+				return err
+			}
+			rpmFull = filepath.Join(rpmPath, rpm)
+			if _, err = os.Stat(rpmFull); err != nil {
+				fmt.Println("rpm not found: ", rpmFull)
+				return err
+			}
+		}
+	}
+
+	return extractRpm(chrootDir, rpmFull, 5)
```

Comment:
I agree. I would be better to retry the whole extract step and let the caller decide on the number of retries.

Reply:
should I reuse the retires flag, or implement a new flag?

Reply:
I would suggest using the existing retries flag. Its default value is 3. While testing if we feel that value is low, we could use a separate hard-coded value. 
I would like to avoid the user having to choose a value in this case, because this is a workaround we have as part of an implementation. The user might not be aware of why its used unless they understand the internal implementation details.

Reply:
Existing retry flags were intended to be used with download failures, so it makes sense to let the user decide since we don't know how good or bad their network is. What is the need for retries in this case in particular? Why the extraction fails? I don't think we should tie this to the retry flag since they don't seem to serve the same purpose.

Reply:
I agree. I was trying to make a point that lets not have the caller decide the retry value in this case.